building/glass/aptbranch.py

```diff
@@ -73,3 +43,34 @@ def export_key(keyid, armor=False):
     if not result.strip():
         raise Exception("empty result from gpg for keyid: '%s'" % keyid)
     return result
+
+
+class AptBranchConfig:
+    def __init__(self, branch: str):
+        if not os.path.exists(CONFIG_PATH):
+            raise Exception("cannot find apt branch config under %s" % CONFIG_PATH)
+        branches_config = validate.load_validated(CONFIG_PATH, CONFIG_SCHEMA_NAME)
+        self.config = None
+        for branch_config in branches_config["branches"]:
+            if branch_config["name"] == branch:
+                self.config = branch_config
+                break
+
+        if self.config is None:
+            raise Exception("no config found for %s in %s" % (branch, CONFIG_PATH))
```

Comment:
I generally prefer assigning each instance field once during __init__, by convention. I'd have a separate 'config' variable, and only assign it to this.config once it's been fully determined.

Maybe even separate this part out into a separate function.

This is up to your discretion.